File: server/model/system/request/operationRecord.go
package request

import "server/model/common/request"

type orStruct struct {
	Method string // 请求方法
	Path   string // 请求路径
	Status int    // http code
}

// OrSearchParams api分页条件查询及排序结构体
type OrSearchParams struct {
	orStruct
	request.PageInfo
	Asc bool `json:"asc"` // 排序方式:升序true|降序true(默认)
}

File: web/src/api/system/operationRecord.ts
import { request } from "@/utils/service"

export interface OrData {
  ID: number
  CreatedAt: string
  ip: string
  method: string
  path: string
  status: number
  userAgent: string
  reqParam: string
  respData: string
  respTime: number
  userName: string
}

export interface OrDataPageInfo {
  list: OrData[]
  total: number
  page: number
  pageSize: number
}

interface reqOrList extends PageInfo {
  path?: string
  method?: string
  status?: number
  asc?: boolean
}

// 分页获取操作记录
export function getOrListApi(data: reqOrList) {
  return request<ApiResponseData<OrDataPageInfo>>({
    url: "/or/getOrList",
    method: "post",
    data
  })
}

// 删除操作记录
export function deleteOrApi(data: reqId) {
  return request<ApiResponseData<null>>({
    url: "/or/deleteOr",
    method: "post",
    data
  })
}

// 批量删除操作记录
export function deleteOrByIdsApi(data: reqIds) {
  return request<ApiResponseData<null>>({
    url: "/or/deleteOrByIds",
    method: "post",
    data
  })
}
